assignments/lambda-classes.js

```diff
@@ -1 +1,177 @@
 // CODE here for your Lambda Classes
+class Person {
+    constructor({name, age, location, gender}) {
+        this.name = name;
+        this.age = age;
+        this.location = location;
+        this.gender = gender;
+    }
+
+    speak () {
+        return `Hello my name is ${this.name}, I am from ${this.location}`
+    }
+};
+
+
+/**
+ * @class Instructor
+ * @extends {Person}
+ */
+class Instructor extends Person {
+    constructor ({name, age, location, gender, specialty,favLanguage, catchPhrase}){
+        super ({name, age, location, gender});
+        this.specialty = specialty;
+        this.favLanguage = favLanguage;
+        this.catchPhrase = catchPhrase;
+    }
+
+
+    /**
+     * @returns a string of learning subject
+     * @memberOf Instructor
+     */
+    demo (subject) {
+        return `Today we are learning about ${subject}`;
+    }
+
+
+    /**
+     * @param {any} student 
+     * @param {any} subject 
+     * @returns a tring of each students name with a perfect score on a subject
+     * @memberOf Instructor
+     */
+    grade (student, subject) {
+       return `${student.name} recieves a perfect score on ${subject}`;
+    }
+
+    points (max) {
+        let studentGrade = Math.floor(Math.random() * Math.floor(max))
+        if (studentGrade <=75) {
+           return studentGrade += Math.random() * 10
+        } else {
+           return studentGrade -= Math.random() * 1.3
+        }
+    }
+
+}
+
+/**
+ * @class Students
+ * @extends {Person}
+ */
+class Students extends Person {
+    constructor ({name, age, location, gender, previouBackground, className,favSubjects}) {
+        super({name, age, location, gender});
+        this.previouBackground = previouBackground;
+        this.className = className;
+        this.favSubjects = favSubjects;
+        this.grade = Math.floor(Math.random() * Math.floor(100));
+    }
+
+
+    /**
+     * @returns a string listing students favorite subject one by one
+     * @memberOf Students
+     */
+    listsSubjects () {
+        return this.favSubjects.forEach((favSubject, i) => console.log(`${i+1}: ${favSubject};`));
+    }
```

Comment:
Be careful here, the last value is returned from the forEach loop is undefined.

Reply:
My bad, you get `undefined` when you log the function to the console. 
You could want to refactor this function to return a string with all the values of your array.